core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -549,6 +556,49 @@ public ServerBuilder workerGroup(int numThreads) {
         return this;
     }
 
+    /**
+     * Sets the worker {@link EventLoopGroup} which is responsible for performing socket I/O and running
+     * {@link Service#serve(ServiceRequestContext, Request)}.
+     * If not set, {@linkplain CommonPools#workerGroup() the common worker group} is used.
+     *
+     * @param workerGroup the worker {@link EventLoopGroup}
+     * @param shutdownOnStop whether to shut down the worker {@link EventLoopGroup}
+     *                       when the {@link Server} stops
+     * @param gracefulShutdown the {@link GracefulShutdown} configuration to be used when the worker
+     * {@link EventLoopGroup} is being shut down, makes sense only if {@link #shutdownWorkerGroupOnStop} is true
+     */
+    @UnstableApi
+    public ServerBuilder workerGroup(EventLoopGroup workerGroup, boolean shutdownOnStop, GracefulShutdown gracefulShutdown) {
```

Comment:
On second thought, it probably makes sense that `shutdownOnStop` isn't accepted at all since setting `gracefulShutdown` implies that the `workerGroup` will be shutdown. (it may be nice to indicate this in the javadocs though) Let me know what you think.

```suggestion
    public ServerBuilder workerGroup(EventLoopGroup workerGroup, GracefulShutdown gracefulShutdown) {
```

Ditto for the equivalent method in `ClientFactoryBuilder`

Reply:
Implemented.